src/mito2/src/access_layer.rs

```diff
@@ -470,7 +470,7 @@ impl TempFileCleaner {
     /// Removes the SST and index file from the local atomic dir by the file id.
     pub(crate) async fn clean_by_file_id(&self, file_id: FileId) {
         let sst_key = IndexKey::new(self.region_id, file_id, FileType::Parquet).to_string();
-        let index_key = IndexKey::new(self.region_id, file_id, FileType::Puffin).to_string();
+        let index_key = IndexKey::new(self.region_id, file_id, FileType::Puffin(0)).to_string(); // TODO(discord9): confirm correctness
```

Comment:
We may also have to call TempFileCleaner when the `IndexBuildTask` fails to build an index.

Reply:
> We may also have to call TempFileCleaner when the `IndexBuildTask` fails to build an index.

added a `do_abort_clean_fs_temp_dir` for `Indexer`